docs/proposals/caching/README.md

```diff
@@ -0,0 +1,254 @@
+type ResourceSelector struct {
+	// APIVersion represents the API version of the target resources.
+	// +required
+	APIVersion string `json:"apiVersion"`
+
+	// Kind represents the Kind of the target resources.
+	// +required
+	Kind string `json:"kind"`
+
+	// Namespace of the target resource.
+	// Default is empty, which means all namespaces.
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
+
+	// Indexes is a list of custom index
+	// Default is empty, which means no custom index.
+	// +optional
+	Indexes []ResourceIndexRef `json:"indexes,omitempty"`
+}
+
+// ResourceIndexRef contains a reference to the kubernetes resource index
+// +enum
+type ResourceIndexRef struct {
+	// +required
+	Name  string   `json:"name" protobuf:"bytes,1,opt,name=name"`
+	Kinds []string `json:"kinds" protobuf:"bytes,2,opt,name=kinds"`
+}
+
+var (
+	LabelIndex    = ResourceIndexRef{Name: "label", Kinds: []string{"*"}}
+	NodeNameIndex = ResourceIndexRef{Name: "nodeName", Kinds: []string{"Pod"}}
+)
+
+// ClusterCacheStatus defines the observed state of ClusterCache
+type ClusterCacheStatus struct {
+	// +optional
+	Resources []ResourceStatusRef `json:"resources,omitempty"`
+	// +optional
+	StartTime *metav1.Time `json:"startTime,omitempty"`
+}
+
+type ResourceStatusRef struct {
+	// +required
+	Cluster string `json:"cluster"`
+	// +optional
+	APIVersion string `json:"apiVersion,omitempty"`
+	// +required
+	Kind string `json:"kind"`
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
+	// +required
+	State CachePhase `json:"state"`
+	// +required
+	TotalNum int32 `json:"totalNum"`
+	// +required
+	UpdateTime *metav1.Time `json:"updateTime"`
+}
+
+// CachePhase is the current state of the cache
+// +enum
+type CachePhase string
+
+// These are the valid statuses of cache.
+const (
+	CacheRunning CachePhase = "Running"
+	CacheFailed  CachePhase = "Failed"
+	CacheUnknown CachePhase = "Unknown"
+)
+
+type ResourceStateRef struct {
+	// +required
+	Phase CachePhase `json:"phase"`
+	// +optional
+	Reason string `json:"reason"`
+}
+
+//+kubebuilder:object:root=true
+
+// ClusterCacheList contains a list of ClusterCache
+type ClusterCacheList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []ClusterCache `json:"items"`
+}
+
+func init() {
+	SchemeBuilder.Register(&ClusterCache{}, &ClusterCacheList{})
+}
+```
+
+#### Example
+
+The following example shows how to create a ClusterCache CRD.
+
+```yaml
+apiVersion: clustercaches.karmada.io/v1alpha1
+kind: ClusterCache
+metadata:
+  name: clustercache-sample
+spec:
+  clusterSelectors:
+    - clusterNames:
+        - member1
+        - member2
+        - member3
+  resourceSelectors:
+    - kind: Pod
+    - kind: Ingress
+      apiVersion: networking.k8s.io/v1
+    - kind: DaemonSet
+      namespace: kube-system
+    - kind: Deployment
+      indexes:
+        - name: label
+          kinds:
+            - "*"
```

Comment:
The worlds `kind` and `Deployment` will be present twice if the index reference is restricted to Deployment.
It seems a little bit redundant and not elegant. Isn't?

```yaml
    - kind: Deployment
      indexes:
        - name: label
          kinds:
            - "Deployment"
```

Reply:
Removed it.